src/util/accessibility/live.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable, Inject, OnDestroy} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+
+
+
+@Injectable()
+export class Live implements OnDestroy {
+  private _element: HTMLElement;
+
+  constructor(@Inject(DOCUMENT) private _document: any) { this._element = this._createElement(); }
+
+  ngOnDestroy() { this._element.remove(); }
+
+  async say(message: string): Promise<void> {
+    const element = this._element;
+
+    element.textContent = '';
+    await new Promise(resolve => setTimeout(resolve, 100));
```

Comment:
Why `100`? How was this number chosen? Why does it need to be async in the first place? As a very minimum we need documentation that explain this.

Reply:
This is all explained there: [material2/live-announcer.ts at 6405da9b8e8532a7e5c854c920ee1815c275d734 · angular/material2](https://github.com/angular/material2/blob/6405da9b8e8532a7e5c854c920ee1815c275d734/src/cdk/a11y/live-announcer/live-announcer.ts#L50).

It seems natural to wait a bit in case, knowing the unpredictability of screen readers. However, choosing a proper amount of delay is purely arbitrary.